apps/src/templates/sectionProgress/progressTables/ProgressTableLevelBubble.jsx

```diff
@@ -0,0 +1,180 @@
+import React from 'react';
+import Radium from 'radium';
+import PropTypes from 'prop-types';
+import FontAwesome from '@cdo/apps/templates/FontAwesome';
+import i18n from '@cdo/locale';
+import * as progressStyles from '@cdo/apps/templates/progress/progressStyles';
+import color from '@cdo/apps/util/color';
+import {LevelStatus} from '@cdo/apps/util/sharedConstants';
+
+const styles = {
+  container: {
+    ...progressStyles.flex,
+    width: progressStyles.BUBBLE_CONTAINER_WIDTH
+  },
+  main: {
+    ...progressStyles.flex,
+    ...progressStyles.font,
+    boxSizing: 'content-box',
+    width: progressStyles.DOT_SIZE,
+    height: progressStyles.DOT_SIZE,
+    borderRadius: progressStyles.DOT_SIZE,
+    fontSize: 16,
+    letterSpacing: -0.11,
+    margin: '3px 0px',
+    position: 'relative'
+  },
+  largeDiamond: {
+    width: progressStyles.DIAMOND_DOT_SIZE,
+    height: progressStyles.DIAMOND_DOT_SIZE,
+    borderRadius: 4,
+    transform: 'rotate(45deg)',
+    margin: '6px 0px'
+  },
+  small: {
+    ...progressStyles.inlineBlock,
+    width: progressStyles.LETTER_BUBBLE_SIZE,
+    height: progressStyles.LETTER_BUBBLE_SIZE,
+    borderRadius: progressStyles.LETTER_BUBBLE_SIZE,
+    lineHeight: '12px',
+    fontSize: 12,
+    margin: progressStyles.LETTER_BUBBLE_MARGIN,
+    paddingRight: progressStyles.LETTER_BUBBLE_PAD,
+    paddingBottom: progressStyles.LETTER_BUBBLE_PAD,
+    textAlign: 'center',
+    verticalAlign: 'middle'
+  },
+  contents: {
+    whiteSpace: 'nowrap',
+    lineHeight: '16px'
+  },
+  diamondContents: {
+    // undo the rotation from the parent
+    transform: 'rotate(-45deg)'
+  },
+  bonusDisabled: {
+    backgroundColor: color.lighter_gray,
+    color: color.white
+  },
+  unpluggedContainer: {
+    ...progressStyles.flex,
+    borderRadius: 20,
+    padding: '6px 10px',
+    margin: '3px 0px',
+    position: 'relative'
+  },
+  unpluggedText: {
+    ...progressStyles.font,
+    fontSize: 12,
+    letterSpacing: -0.12
+  }
+};
+
+class ProgressTableLevelBubble extends React.PureComponent {
```

Comment:
I'm not sure what the best way or place to do this, but a header comment describing that this component and how different values of levelStatus, levelKind, etc. affect the size, shape, and color of the component might be useful?

Reply:
i would prefer to defer this until we consolidate all our progress components in H1